website/frontend/src/tests/FormPayload/FormPayload.test.tsx

```diff
@@ -0,0 +1,500 @@
+    });
+
+  const verifyValues = (formValues: FormValues) => waitFor(() => expect(submitCallbackMock).toBeCalledWith(formValues));
+
+  afterEach(() => {
+    cleanup();
+    jest.resetAllMocks();
+  });
+
+  it('should displayed manual textarea', () => {
+    const { rerender } = render(<TestFromPayload onSubmit={jest.fn()} />);
+
+    const manualTextarea = screen.getByPlaceholderText('// Enter your payload here');
+
+    expect(screen.queryByLabelText('Manual input')).toBeNull();
+    expect(manualTextarea).toBeInTheDocument();
+    expect(manualTextarea).toHaveTextContent('');
+    expect(screen.queryByText('Select file')).toBeNull();
+
+    rerender(<TestFromPayload values={INPUT_PAYLOAD_VALUES} onSubmit={jest.fn()} />);
+
+    expect(screen.getByLabelText('Manual input')).toBeInTheDocument();
+
+    expect(manualTextarea).not.toBeInTheDocument();
+    expect(screen.getByRole('checkbox')).not.toBeChecked();
+    expect(screen.queryByText('Select file')).toBeNull();
+
+    expect(screen.getAllByRole('group')).toHaveLength(2);
+    expect(screen.getAllByRole('combobox')).toHaveLength(2);
+
+    toggleView();
+
+    expect(screen.getByText('Select file')).toBeInTheDocument();
+  });
+
+  it('should submit correct manual', async () => {
+    render(
+      <TestFromPayload
+        values={INPUT_PAYLOAD_VALUES}
+        onSubmit={jest.fn((values: FormValues) => submitCallbackMock(values))}
+      />
+    );
+
+    toggleView();
+
+    const manualTextbox = screen.getByRole('textbox');
+
+    expect(manualTextbox).toHaveValue(getPreformattedText(INPUT_MANUAL_PAYLOAD));
+
+    const manualText = `
+    {
+      "Test1": {
+          "firstName": "first",
+          "secondName": "second",
+          "age": null,
+      }
+    }`;
+
+    changeFieldValue(manualTextbox, manualText);
+
+    expect(manualTextbox).toHaveValue(manualText);
+
+    submit();
+    await verifyValues({ payload: manualText });
+  });
+
+  it('should set form value from json', async () => {
+    const { container } = render(
+      <TestFromPayload
+        values={INPUT_PAYLOAD_VALUES}
+        onSubmit={jest.fn((values: FormValues) => submitCallbackMock(values))}
+      />
+    );
+
+    toggleView();
+
+    const textbox = screen.getByRole('textbox');
+
+    const fileInput = container.getElementsByTagName('input')[1];
+
+    expect(fileInput).toBeInTheDocument();
+    expect(screen.getByText('Select file')).toBeInTheDocument();
+
+    //upload file
+
+    const testFile = new File([INPUT_FILE_CONTENT], 'test.json', { type: FILE_TYPES.JSON });
+
+    act(() => {
+      fireEvent.change(fileInput, {
+        target: { files: [testFile] },
+      });
+    });
+
+    await waitFor(() => expect(screen.queryByText('Select file')).toBeNull());
+
+    expect(screen.getByText('test.json')).toBeInTheDocument();
+
+    await waitFor(() => expect(textbox).toHaveValue(INPUT_FILE_CONTENT));
+
+    submit();
+    await verifyValues({ payload: INPUT_FILE_CONTENT });
+
+    //drop file
```

Comment:
```suggestion
    // remove file
```
drop in file input context resonates with drag'n'drop somehow